src/Core/View.js

```diff
@@ -165,12 +151,6 @@ function _preprocessLayer(view, layer, provider, parentLayer) {
                 providerPreprocessing = Promise.resolve();
             }
         } else if (layer.source) {
-            if (!layer.source.isSource) {
```

Comment:
`else if (layer.source)` is necessary? may be replace by simple `else`.

Reply:
I let it as it is, but removed the `|| providerPreprocessing`